community/kernel-api/src/main/java/org/neo4j/internal/kernel/api/Transaction.java

```diff
@@ -22,23 +22,31 @@
 /**
  * A transaction with the graph database.
  */
-public interface Transaction
+public interface Transaction extends AutoCloseable
 {
+    enum Type
+    {
+        implicit,
+        explicit
+    }
+
     /**
      * The store id of a read-only transaction.
      */
     long READ_ONLY_TRANSACTION = 0;
 
     /**
-     * Commit this transaction.
-     * @return the id of the commited transaction, or {@link READ_ONLY_TRANSACTION} if this was a read only transaction.
+     * Marks this transaction as successful. When this transaction later gets {@link #close() closed}
+     * its changes, if any, will be committed. If this method hasn't been called or if {@link #failure()}
+     * has been called then any changes in this transaction will be rolled back as part of {@link #close() closing}.
      */
-    long commit();
+    void success();
```

Comment:
Looks like the `READ_ONLY_TRANSACTION` constant is no longer used. We already have `READ_ONLY` (and `ROLLBACK`) in `KernelTransaction` anyway.

But supposedly the transaction id is still useful, right? I'm thinking that maybe this interface shouldn't be `AutoCloseable` afterall.

Reply:
Yes you are right, I merged the constants.

The transaction ID seems only to be used by `KernelTransaction` users via `KernelTransaction.CloseListener` (which is implemented internally to unbind a transaction from it's Thread), via `KernelTransaction.getTransactionId()` which is used to implement the public `org.neo4j.graphdb.event.TransactionData` interface, and via `KernelTransaction.closeTransaction()` which is not used apart from a few tests. In the first two cases we are not in the position to actually close the transaction, so having the `long commit()` method would not help. In the last case we are actually given this exact functionality, but it isn't used. Let's keep the AutoCloseable approach for now?